apps/src/code-studio/assets/SpriteUpload.jsx

```diff
@@ -33,7 +53,11 @@ export default class SpriteUpload extends React.Component {
     uploadStatus: {
       status: UploadStatus.none,
       message: ''
-    }
+    },
+    extensionError: TestStatus.pending,
+    largeImageError: TestStatus.pending,
+    smallImageWarning: TestStatus.pending,
+    filenameError: TestStatus.pending
```

Comment:
Looking at this set of states, I'm wondering if we might be duplicating information. It looks like we have "filenameOverride', and 'badFile'," in UploadStatus and 'badFile' covers the same cases as 'filenameError'. Could we find a way to drop one of these? Maybe pull 'filenameOverride' and 'badFile' from UploadStatus - so that it purely covers the success/failure of uploading to S3?

As I was writing that, I also realized that this is a huge state and this file is huge! Can we break out the image-selection-portion into it's own component? It could handle the image checks and then callback a single 'status' into this function. That would help break up some of this code into more consumable pieces. Happy to hop on a call and brainstorm!

Reply:
Great suggestion about breaking out image-selection into a separate component! Would definitely help manage the tracking of state and reduce duplication.